src/utils.jl

```diff
@@ -123,9 +123,12 @@ end
 
 function scopename(tn::TypeName)
     modpath = Base.fullname(tn.module)
-    return Expr(:., _scopename(modpath...), QuoteNode(tn.name))
+    if isa(modpath, Tuple{Symbol})
+        return Expr(:., modpath[1], QuoteNode(tn.name))
+    else
+        return Expr(:., _scopename(modpath...), QuoteNode(tn.name))
+    end
 end
-_scopename(sym) = sym
```

Comment:
Just to check my understanding, does this change reduce a chance of (unused) specialization, right ? (maybe otherwise `_scopename(parent, child, rest...) = Expr(:., parent, _scopename(child, rest...))` can recur back into this method ?)

Reply:
It was more that attempts to call `_scopename(sym)` were being implemented by runtime-dispatch. Now that I think about it, that would also be true of longer module paths: the length of `modpath` is not inferrable. I should rewrite this so it never dispatches to anything. (later, busy with other things now)